### DISCUSSION TITLE
How can i deploy updates one at a time?

### DISCUSSION THREAD

**Discussion by wisquimas** [containrrr/watchtower#1437] [General]
Hi, guys.
I have a server with multiple containers with the same image.
When watchtower see that there is an update do the next:

1. Download the new image.
2. stop 1 by one the containers with that image
3. Wait until all the containers are down
4. Redeploy all de containers.
5. (I dont see when the system deletes the images with the --cleanup argument... so, is the 5th xD)

OK, I wanna change that to:
1. Dowload the new image.
2. Stop one container and redeploy with new image.
3. Do that one by one. 

I'm thinking in use two tags to handle that in a custom code but i don't know if you can now do that.

If there is an configuration to do that, can you tell where is the documentation of that? (or an example)
If not is posible now, can you tell me where in watchtower code is the file and line where you use the callback after the detection of the new image updated?

Sorry for my english :(

**Comment by piksel** [comment]
I don't think that's possible without #1231.
Or, at least not without a complex set of [lifecycle hooks](https://containrrr.dev/watchtower/lifecycle-hooks/).

> where in watchtower code is the file and line where you use the callback after the detection of the new image updated?

It's not that simple, but the file you are looking for is probably 
[/internal/actions/update.go](https://github.com/containrrr/watchtower/blob/9ea5d3b5e8971b5dca72cb18d7291e7136321451/internal/actions/update.go)
